sandbox/apparmor/apparmor.go

```diff
@@ -478,9 +494,14 @@ func AppArmorParser() (cmd *exec.Cmd, internal bool, err error) {
 }
 
 // tryAppArmorParserFeature attempts to pre-process a bit of apparmor syntax with a given parser.
-func tryAppArmorParserFeature(cmd *exec.Cmd, rule string) bool {
+func tryAppArmorParserFeature(cmd *exec.Cmd, flags []string, rule string) bool {
 	cmd.Args = append(cmd.Args, "--preprocess")
-	cmd.Stdin = bytes.NewBufferString(fmt.Sprintf("profile snap-test {\n %s\n}", rule))
+	flagSnippet := ""
+	if len(flags) > 0 {
+		flagSnippet = fmt.Sprintf("flags=(%s)", strings.Join(flags, ","))
+	}
+	cmd.Stdin = bytes.NewBufferString(fmt.Sprintf("profile snap-test %s {\n %s\n}",
```

Comment:
tiny nitpick: this would prevent adding two spaces if there are no flags. Then `apparmor_test.go` wouldn't need the whitespace changes

```suggestion
		flagSnippet = fmt.Sprintf("flags=(%s) ", strings.Join(flags, ","))
	}
	cmd.Stdin = bytes.NewBufferString(fmt.Sprintf("profile snap-test %s{\n %s\n}",
```